index.js

```diff
@@ -1,216 +1,413 @@
 /*
+
  * WebSSH2 - Web to SSH2 gateway
+
  * Bill Church - https://github.com/billchurch - April 2016
+
  *
+
  */
+
 var express = require('express');
+
 var app = express();
+
 var cookieParser = require('cookie-parser');
+
 var server = require('http').Server(app);
+
 var io = require('socket.io')(server, {
+
     path: '/webssh/socket.io'
+
 });
+
 var path = require('path');
+
 var fs = require('fs');
 
+
+
 var basicAuth = require('basic-auth');
+
 var ssh = require('ssh2');
+
 var readConfig = require('read-config'),
+
     config = readConfig(__dirname + '/config.json');
+
 var myError = " - ";
+
 var serverStatusFile = "/appsvctmp/status.txt";
 
+
+
 function logErrors(err, req, res, next) {
+
     console.error(err.stack);
+
     next(err);
+
 }
 
+
+
 server.listen({
+
     host: config.listen.ip,
+
     port: config.listen.port
+
 }).on('error', function (err) {
+
     if (err.code === 'EADDRINUSE') {
+
         config.listen.port++;
+
         console.log('Address in use, retrying on port ' + config.listen.port);
+
         setTimeout(function () {
+
             server.listen(config.listen.port);
+
         }, 250);
+
     }
+
 });
 
+
+
 app.use(express.static(__dirname + '/webssh/public')).use(function (req, res, next) {
-    /*var myAuth = basicAuth(req);
-    if (myAuth === undefined) {
-        res.statusCode = 401;
-        res.setHeader('WWW-Authenticate', 'Basic realm="WebSSH"');
-        res.end('Username and password required for web SSH service.');
-    } else if (myAuth.name == "") {
-        res.statusCode = 401
-        res.setHeader('WWW-Authenticate', 'Basic realm="WebSSH"');
-        res.end('Username and password required for web SSH service.');
-    } else {*/
-    config.user.name = 'root';
-    config.user.password = 'Docker!';
+
     if (req.originalUrl != null && req.originalUrl.includes("webssh/host")) {
+
         if (req.query.debugconsolereq != null) {
+
+            // Use the runtime User for Kudu, and the dynamically generated password during the startup
+            config.user.name = process.env.USER_NAME;
+
+            config.user.password = process.env.USER_PASSWORD.trim();
```

Comment:
spaces are usually considered reasonable password characters. Is there a reason we are trimming this and not USER_NAME?

Reply:
This is just for Kudu Password, which is controlled by the startup script (generated by us). The main app container honors any character entered by the user